sklearn/naive_bayes.py

```diff
@@ -351,9 +351,10 @@ def _partial_fit(self, X, y, classes=None, _refit=False,
         unique_y_in_classes = in1d(unique_y, classes)
 
         if not np.all(unique_y_in_classes):
+            not_unique_y_in_classes = ~unique_y_in_classes
             raise ValueError("The target label(s) %s in y do not exist in the "
                              "initial classes %s" %
-                             (y[~unique_y_in_classes], classes))
+                             (y[np.where(not_unique_y_in_classes)], classes))
```

Comment:
Yeah, the if statement checks whether all the elements in the array evaluates to true, if not an error is raised. So, instead of using "~unique_y_in_classes" (a boolean array), i've replaced it with np.where , which returns the indices of the non-zero elements in "~unique_y_in_classes"